src/lib/tools/fetchUserArtistFollows.ts

```diff
@@ -0,0 +1,73 @@
+import { tool } from "ai"
+import { extractNodes, metaphysics } from "../metaphysics"
+import { z } from "zod"
+
+const DEFAULT_NUMBER_OF_ARTISTS = 20
+
+export const fetchUserArtistFollows = tool({
+  description: `
+    Fetch a list of artists that the current user is following.
+  `,
+
+  parameters: z.object({
+    size: z
+      .number()
+      .max(50)
+      .default(DEFAULT_NUMBER_OF_ARTISTS)
+      .describe(
+        `The maximum number of artists to return, default to ${DEFAULT_NUMBER_OF_ARTISTS}`
+      ),
+  }),
+
+  execute: async (args) => {
+    const accessToken = process.env.USER_ACCESS_TOKEN
+    const size = args.size ?? DEFAULT_NUMBER_OF_ARTISTS
+
+    let artistFollows: unknown[] = []
+    let lastCursor = null
+
+    while (artistFollows.length < size) {
```

Comment:
Building in support for cursor-based pagination so that we can fetch multiple pages of results if needed (still governed by whatever `max` we set, e.g. L15 above)